Use native-tls/native-certs features of ureq crate

lazareviczoran opened the pull request:
Hello!

We use this crate on one of our projects, once we bumped it to version 0.10.7, we started getting failures for the license checks in cargo-deny, since openblas-build now uses `ureq` for [downloading the openblas source](https://github.com/blas-lapack-rs/openblas-src/blob/master/openblas-build/src/download.rs#L16), which brings in a dependency with a MPL-2.0 license to the Cargo.lock file (it has been brought up in ureq repo as well https://github.com/algesten/ureq/issues/478). 

`ureq` does have some feature flags that provide the possibility to use native-tls, the lib doesn't implement the default tls config when using `native-tls` feature flag (https://github.com/algesten/ureq/blob/main/src/lib.rs#L377-L397), but it allows to setup an agent explicitly by providing the tls config (as it is done in this test https://github.com/algesten/ureq/blob/main/src/lib.rs#L583-L595).
There was a proposal for adding feature flag in `ureq` that includes `rustls` with the `native-certs` flag so that the default agent can work with the `native-certs` (https://github.com/algesten/ureq/pull/482), but it was rejected, so currently it is not possible to just tweak the feature flags on `ureq` and exclude the unwanted sub-dependencies. 

The idea of this PR is to add feature flags to `openblas-src` and `openblas-build` that allows users to choose which tls config they want to use. This PR introduces 2 feature flags (`tls` which includes the default tls that `ureq` uses, and `native-certs` which uses `native-tls` and `native-certs` feature flags in `ureq`).
The default flags are set to use the `tls` feature flag, but worth to note is that this would probably be a breaking change when the default features are disabled and neither of these flags are enabled.

Any thoughts on this?
I look forward to you feedback!

termoshtt commented:
Thanks PR!

Using `native-certs` crate looks better for keeping dependencies list simple.

> The default flags are set to use the tls feature flag, but worth to note is that this would probably be a breaking change when the default features are disabled and neither of these flags are enabled.

To avoid this issue, we should use only `native-certs` case without introducing the feature flag.

lazareviczoran commented:
> Thanks PR!
> 
> Using `native-certs` crate looks better for keeping dependencies list simple.
> 
> > The default flags are set to use the tls feature flag, but worth to note is that this would probably be a breaking change when the default features are disabled and neither of these flags are enabled.
> 
> To avoid this issue, we should use only `native-certs` case without introducing the feature flag.

Thanks for the feedback 🙌 
I've removed the introduced feature flags and updated the PR to use native-certs by default as you suggested.

termoshtt commented:
0.10.8 has been released with this PR https://github.com/blas-lapack-rs/openblas-src/releases/tag/openblas-src-v0.10.8
